new-log-viewer/src/services/LogFileManager.ts

```diff
@@ -230,6 +249,78 @@ class LogFileManager {
         };
     }
 
+    /**
+     * Searches for log events based on the given search string.
+     *
+     * @param searchString The search string.
+     * @param isRegex Whether the search string is a regular expression.
+     * @param matchCase Whether the search is case-sensitive.
+     * @return An object containing the search results.
+     */
+    startQuery (searchString: string, isRegex: boolean, matchCase: boolean): void {
+        this.incrementQueryId();
+
+        // If the search string is empty, or there are no logs, return
+        if ("" === searchString) {
+            return;
+        } else if (0 === this.#numEvents) {
+            return;
+        }
+
+        // Construct search RegExp
+        const regexPattern = isRegex ?
+            searchString :
+            searchString.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
+        const regexFlags = matchCase ?
+            "" :
+            "i";
+        const searchRegex = new RegExp(regexPattern, regexFlags);
+        this.#searchChunk(this.#queryId, 0, searchRegex);
+    }
+
+    /**
+     * Searches for log events in the given range.
+     *
+     * @param queryId
+     * @param beginSearchIdx The beginning index of the search range.
+     * @param searchRegex The regular expression to search
+     * @return
+     */
+    #searchChunk (queryId: number, beginSearchIdx: number, searchRegex: RegExp): void {
+        if (queryId !== this.#queryId) {
+            return;
+        }
+
+        const endSearchIdx = Math.min(beginSearchIdx + SEARCH_CHUNK_SIZE, this.#numEvents);
+        const results: ChunkResults = {};
+
+        for (let eventIdx = beginSearchIdx; eventIdx < endSearchIdx; eventIdx++) {
+            const contentString = this.#decoder.decode(eventIdx, eventIdx + 1)?.[0]?.[0] || "";
```

Comment:
This looks a bit strange. Shouldn't we decide a range and process the results in a batch? Note you can also obtain the log event index from each result (an array of 4 values).